Cabal/Distribution/Utils/LogProgress.hs

```diff
@@ -54,7 +55,7 @@ runLogProgress verbosity (LogProgress m) =
       }
     step_fn :: LogMsg -> NoCallStackIO a -> NoCallStackIO a
     step_fn doc go = do
-        putStrLn (render doc)
+        hPutStrLn stderr (render doc)
```

Comment:
Done.

Reply:
Reverted this particular change.  Changing `runLogProgress.step_fn` to use `notice` instead of `hPutStrLn` breaks too many tests as `notice` will wrap the messages with `-----BEGIN CABAL OUTPUT-----` markers, causing them to get caught in the test output.